internal/slicer/slicer_test.go

```diff
@@ -248,9 +252,13 @@ var slicerTests = []slicerTest{{
 }, {
 	summary: "Copyright is installed",
 	slices:  []setup.SliceKey{{"test-package", "myslice"}},
-	pkgs: map[string][]byte{
-		// Add the copyright entries to the package.
-		"test-package": testutil.MustMakeDeb(append(testutil.TestPackageEntries, testPackageCopyrightEntries...)),
+	archives: map[string]*testArchive{
+		"ubuntu": {
```

Comment:
We need to think a little bit more about the testing strategy. For example, we are using the default `chisel.yaml` so as to avoid defining the archives and thinking about them, but now I have to use `ubuntu` when defining the packages with no indication of where it comes from.

Reply:
True. I would prefer your changes in [`slicer_test.go`](https://github.com/canonical/chisel/pull/142/files#diff-5a7910fb12530b5a3e1772624bae4c4554a531aedc73835ef562b53372a880a5) in canonical#142, with the introduction of `testutil.TestPackage` and `testutil.TestArchive` types.

Considering the diffs are huge, I would rather wait to be honest until that one gets merged to apply the same changes here.

Reply:
The `TestPackage` would have a new `Archive` field:

```go
type TestPackage struct {
	Name    string
	Version string
	Hash    string
	Arch    string
	Data    []byte
	Archive []string
}
```

So the slicer tests would look something like the following:
```go
pkgs: []testutil.TestPackage{{
		Name:     "test-package",
		Hash:     "h1",
		Version:  "v1",
		Arch:     "a1",
		Data:     // Some data,
		Archives: []string{"foo", "bar"}
	}, {
		Name:     "test-package",
		Hash:     "h2",
		Version:  "v1",
		Arch:     "a1",
		Data:     // Different data than above,
		Archives: []string{"jammy"}
	},
},
```

The rationale is -

* We don't have to map by _default_ archive names, which we shouldn't bother to know.
* The packages will be a slice because we want to include same package names with different data and metadata.